feg/gateway/gtp/gtp_client.go

```diff
@@ -122,9 +125,34 @@ func (c *Client) Run(ctx context.Context) error {
 			return
 		}
 	}()
+	glog.V(2).Infof("Wait until local gtp client is ready")
+	waitUntilClientIsReady(c, 0)
 	return nil
 }
 
+// waitUntilClientIsReady is a hack to know when the client is ready and avoid null pointer issues using
+// the GTP-C client too early. Since go-gtp doesn't offer any visibuility on the readines of the connection
+// we use LocalAddrs as indicator
+func waitUntilClientIsReady(c *Client, count int) {
+	defer func() {
+		if count > 50 {
+			time.Sleep(time.Millisecond * 20)
+		}
+		if count > 100 {
+			glog.Errorf("Couldnt start GTP-Client")
+			return
+		}
+		if r := recover(); r != nil {
+			fmt.Print(".")
+			waitUntilClientIsReady(c, count+1)
+		}
+	}()
+	// this call will panic while client is starting
+	addr := c.LocalAddr().String()
+	fmt.Println()
+	glog.V(2).Infof("Started GTP-C client in %s", addr)
+}
+
```

Comment:
This is just a matter of preference, but I think recursively calling the function is a bit confusing to read. Could we just unroll the retries into a loop?

```
func (c *Client) waitUntilReady(ctx context.Context, timeout int) error {
	while true {
		ready := make(chan bool)
		go func() {
			defer func() {
				if r := recover(); r != nil {
					fmt.Print(".")
					ready <- false
				} else {
					ready <- true
				}
			}()
			// this call will panic while client is starting
			addr := c.LocalAddr().String()
			glog.V(2).Infof("Started GTP-C client in %s", addr)
		}()
		if <-ready {
			return nil
		}
		// retry sleeps, timeout logic etc.
	}
	return errors.New("Timed out")
}
```

Reply:
This is a function that will be removed soon once I fix the Go-GTP
For simplicity i prefer to keep it small codewise